### PR TITLE
[WIP] upgrade: tokio 0.3

### LINKED ISSUE #7984: upgrade to tokio 1.0
https://github.com/tokio-rs/tokio/releases/tag/tokio-0.3.0

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes #7984

### CODE REVIEW

File: cli/repl.rs [hunk 1]
```diff
@@ -133,7 +133,7 @@ async fn post_message_and_poll(
         // A zero delay is long enough to yield the thread in order to prevent the loop from
         // running hot for messages that are taking longer to resolve like for example an
         // evaluation of top level await.
-        tokio::time::delay_for(tokio::time::Duration::from_millis(0)).await;
+        tokio::time::sleep(tokio::time::Duration::from_millis(0)).await;
```
Comment (a1phyr) [inline cli/repl.rs:136]: Unrelated to tokio 0.3 upgrade, but I think using `tokio::task::yield_now` would be better here.

Comment (bartlomieju) [reply cli/repl.rs:136]: CC @caspervonb

Comment (caspervonb) [reply cli/repl.rs:136]: As far as I remember I tried that early and it spun hot at 100% but can take a look at it again.

Comment (caspervonb) [reply cli/repl.rs:136]: Yeah it'll turn hot.

Screenshot 2020-10-20 at 8 47 03 PM

**Comment by bartlomieju** [comment]
We need to do a lot of refactors before migration to Tokio 0.3 becomes feasible. Closing for now.

**Pull request closed without merging**